fix: Message forwards do not require other fields (#7830)
hammerandchisel/discord-api-docs: docs/resources/message.mdx

###### JSON/Form Params

:::info

> When creating a message, apps must provide a value for **at least one of** `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, `message_reference` or `poll`.

Reviewer: Thanks for the PR. Two things:

**One** 

I was testing this out, because I wanted to see how it works. `message_reference` can't be sent on it's own if it's a FORWARD (1) request, but not a DEFAULT (0). If it's a DEFAULT (reply), it has to have some of the other values.

I'm wondering if we should make this something like:

```suggestion
When creating a message, apps must provide a value for **at least one of** `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, forwarding `message_reference` or `poll`.
```

I don't _love_ it, but I can't think of anything better without writing a whole paragraph. WDYT?

Since if you try and send a `message_reference` of type `REPLY`  on it's own without any other values, you get back `{"message": "Cannot send an empty message", "code": 50006}`.

**Two** 

We should also update line 644, since it duplicates this one:

> \* At least one of `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, or `poll` is required.

Also while we're at it - we should ensure the `*` is actually on all the things we reference there (I noticed it's missing on "poll" and also `message_reference`.

Thanks!

Reviewer 2: what about
```suggestion
When creating a message, apps must provide a value for **at least one of** `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, `message_reference`, `poll` or a [forwarded](add-docs-link-here) message.
```

Reviewer: 🤔 that seems confusing to me, since a `message_reference` is the way to create forwarded message, this reads like a forwarded message is something else.

Also, a forwarded message through the `message_reference` is (afaik) the only way to **only** have a `message_reference` as the only specified value in that list (but that is not true for a reply), so saying you can have it as the only value is not always correct.

i.e. this works:

```shell
curl -X POST "https://discord.com/api/v10/channels/$CHANNEL_ID/messages" \
  -H "Authorization: Bot $BOT_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "message_reference": {
      "message_id": "$MESSAGE_ID",
      "channel_id": "$CHANNEL_ID",
      "guild_id": "$GUILD_ID",
      "type": 1
    }
  }'
  ```

But this doesn't because you can't reply to a message with an empty message body.
  ```shell
curl -X POST "https://discord.com/api/v10/channels/$CHANNEL_ID/messages" \
  -H "Authorization: Bot $BOT_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "message_reference": {
      "message_id": "$MESSAGE_ID",
      "channel_id": "$CHANNEL_ID",
      "guild_id": "$GUILD_ID",
      "type": 0
    }
  }'
  ```

Reviewer 2: ah yeah true. I forgot that forward doesn't has a separate topic hmm

Reviewer: Yeah, it's a bit of a tricky one.

Reviewer 3: The other avenue is to stop the maintenance burden and apparent complexity by generalising:

```suggestion
When creating a message, apps must provide a field that results in visible content.
```

Reviewer 2: That's gonna be too difficult for beginners to understand imo

Reviewer: > That's gonna be too difficult for beginners to understand imo

I'm going to +1 this. I 100% get the impetus, but I'd rather defer on being explicit rather than have people infer the meaning.

Reviewer 2: forwarding `message_reference` sounds like the way to go then for now

Author: updated, i think that'll do

> When creating a message, apps must provide a value for **at least one of** `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, `poll`, or `message_reference` (only when reference type is [`FORWARD`](/docs/resources/message#message-reference-types)).

Reviewer 2: honestly that still sounds off. i was more a fan of marks suggestion

```suggestion
When creating a message, apps must provide a value for **at least one of** `content`, `embeds`, `sticker_ids`, `components`, `files[n]`, [forwarding](/docs/resources/message#message-reference-types) `message_reference` or `poll`.
```

maybe we could meet in the middle here?

Reviewer 2: Also don't forget about 
> We should also update line 644, since it duplicates this one

Author: isn't specifying the type clearer here? not sure tbh

Reviewer 2: it's the wording for me here. it doesn't add up with the rest of the sentence

Author: updated, lmk

Reviewer 2: i wouldn't remove the info box 😅

Author: it doesn't make sense to keep both of them

Reviewer 2: lgtm then

Reviewer: I like that a lot. Nicely done.
